applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1306,12 +1317,58 @@ int unicast_client_stop(void)
 	return 0;
 }
 
+static bool ts_valid(enum timestamp_state ts_state)
+{
+	return (ts_state == TS_VALID);
+}
+
+#define SDU_INTERVAL_US (10000u)
+/**
+ * @brief
+ *
+ * @param tx_info
+ * @param ts_state
+ * @return uint32_t
+ */
+uint32_t timestamp_select(struct bt_iso_tx_info const *const tx_info)
+{
+
+	if (!ts_valid(ts_state[AUDIO_CH_L]) && !ts_valid(ts_state[AUDIO_CH_R])) {
+		LOG_WRN("No valid timestamp found");
+		return 0;
+	}
+
+	if (ts_valid(ts_state[AUDIO_CH_L]) && !ts_valid(ts_state[AUDIO_CH_R])) {
+		return tx_info[AUDIO_CH_L].ts;
```

Comment:
at this point, it should be ok to do `ts_state[AUDIO_CH_R] = TS_VALID`.

Reply:
I see your point, but that would imply that the timestamp we already have in tx_info[AUDIO_CH_R] is valid, which it may not be.

Reply:
I agree that it is not valid at this point in time, but in this state, I believe we can safely anticipate that it will be valid the next time we are here. However, I see that for the purposes of the state machine it more tidy and clear to not change state in here.